src/fpu/fpu.cpp

```diff
@@ -364,14 +364,14 @@ void FPU_ESC3_EA(Bitu rm,PhysPt addr) {
 }
 
 void FPU_ESC3_Normal(Bitu rm) {
-	Bitu group=(rm >> 3) & 7;
-	Bitu sub=(rm & 7);
+	const auto group = static_cast<unsigned>((rm >> 3) & 7);
+	const auto sub = static_cast<unsigned>(rm & 7);
```

Comment:
With the `& 7` capping the result to 7, in theory something like this should be good enough (and safe, too)?

``` c++
unsigned char group = (rm >> 3) & 7;
unsigned char sub = rm & 7;
```

or

``` c++
uint8_t group = (rm >> 3) & 7;
uint8_t sub = rm & 7;
```

I couldn't get this to generate any warnings with gcc, clang, or the analyzer - but I haven't pushed it to coverity. 
I suppose, worst case if coverity doesn't like bit-wising a long with a short, we'd have to chop `rm` down to a short too:

``` c++
uint8_t group = static_cast<uint8_t>(rm >> 3) & 7;
uint8_t sub = static_cast<uint8_t>(rm) & 7;
```

Reply:
I'm probably missing something very subtle though :-)

Reply:
Correct - I started by looking at warning in printf formatting and when I figured out this `& 7` (meaning only 3 bits are used, so whole `Bitu` is unnecessary here) I decided it's safe to implement a fix here.

Decided to use `unsigned`, so:

1. make it less confusing to readers who look at `0x01`, `0x02`, etc below.
2. I could use simple `%u` instead of `PRI<something>` in LOG functions.
3. `static_cast` to prevent any tool from flagging implicit conversion from wider type (Bitu) to smaller one;  `auto` to avoid writing type twice, in case someone will want to change it in the future.

I can resign from `auto`, but I would prefer to keep `static_cast` (just in case any tool will complain) and keep using `unsigned` (or `int`) (although any other integer type would be fine here).

Reply:
Thanks for talking through the reasoning; and I agree that the biggest concern is number 3. 

I hadn't though of using `auto` to DRY the type; this is great, and can become a coding standard for us: "if a new variable's value is explicitly cast to a type, then use auto to avoid repeating the type twice"

 (normally I'm not a fan of auto if the type is totally unknown to the casual reader), but in this case it's a very worthy improvement!